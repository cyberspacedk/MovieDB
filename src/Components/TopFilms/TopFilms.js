import React from 'react';

// eslint-disable-next-line react/prop-types
const TopFilms = ({ topFilms }) => {
  console.log(topFilms);
  const chart = 'cxcx';

  // topFilms.results.map(film => (
  //  <li key={film.id}>
  //    <h2>{film.title}</h2>
  //    <p>{film.overview}</p>
  //  </li>
  // ));

  return <ul>{chart}</ul>;
};

export default TopFilms;
