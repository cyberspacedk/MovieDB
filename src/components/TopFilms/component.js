import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const List = styled.ul`
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  max-width: 950px;
  margin: 0 auto;
`;
const Item = styled.li`
  display: flex;
  flex-direction: column;
  flex-basis: 17%;
  position: relative;
  margin-bottom: 15px;
  h2 {
    font-size: 0.9rem;
    text-align: center;
    background-color: #1d1d1dad;
    padding: 5px;
    color: white;
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    margin: 0;
  }
`;
const Label = styled.h2`
  display: inline-block;
  background-color: green;
  padding: 5px 10px;
  color: #fff;
`;

const TopFilms = ({ loading, topFilms }) => {
  return (
    <div>
      <Label>Top Week chart </Label>
      <List>
        {loading && <li>...Loading</li>}
        {topFilms.results !== undefined &&
          topFilms.results.map(film => (
            <Item key={film.id}>
              <img
                src={`https://image.tmdb.org/t/p/w200${film.poster_path}`}
                alt="Poster"
              />
              <h2> {film.title} </h2>
            </Item>
          ))}
      </List>
    </div>
  );
};

TopFilms.propTypes = {
  loading: PropTypes.bool.isRequired,
  topFilms: PropTypes.shape.isRequired,
};

export default TopFilms;
