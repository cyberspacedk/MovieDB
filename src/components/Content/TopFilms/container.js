/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { fetchDataRequest } from '../../../store/topFilms/actions';
import { singleFilmRequest } from '../../../store/singleFilm/actions';
import TopFilms from './component';
import {
  getTopFilmsSelector,
  isloading,
  isError,
} from '../../../store/topFilms/selectors';

export class TopFilmsContainer extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.fetchDataRequest();
  }

  render() {
    const { error: _e, fetchDataRequest: _f, ...restProps } = this.props;

    return <TopFilms {...restProps} />;
  }
}

export const mstp = state => ({
  topFilms: getTopFilmsSelector(state),
  loading: isloading(state),
  error: isError(state),
});

const mdtp = {
  fetchDataRequest,
  singleFilmRequest,
};

TopFilmsContainer.propTypes = {
  fetchDataRequest: PropTypes.func.isRequired,
  topFilms: PropTypes.arrayOf(PropTypes.shape),
  loading: PropTypes.bool,
  error: PropTypes.bool,
};

export default compose(
  connect(
    mstp,
    mdtp,
  ),
  withRouter,
)(TopFilmsContainer);
