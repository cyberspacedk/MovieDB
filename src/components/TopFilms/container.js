/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { GET_FILMS_PATH } from '../../api/api';
import fetchFilms from '../../Redux/Actions/getTopFilms';
import TopFilms from './component';

import {
  getTopFilmsSelector,
  isloading,
  isError,
} from '../../Redux/Selectors/getTopFilmsSelector';

class TopFilmsContainer extends Component {
  componentDidMount() {
    this.props.getFilms(GET_FILMS_PATH);
  }

  render() {
    return <TopFilms {...this.props} />;
  }
}

const mstp = state => ({
  topFilms: getTopFilmsSelector(state),
  loading: isloading(state),
  error: isError(state),
});

const mdtp = {
  getFilms: fetchFilms,
};

TopFilmsContainer.propTypes = {
  getFilms: PropTypes.func.isRequired,
};

export default connect(
  mstp,
  mdtp,
)(TopFilmsContainer);
