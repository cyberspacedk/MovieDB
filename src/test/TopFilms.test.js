import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TopFilms from '../components/TopFilms/component';

describe('Component: TopFilms', () => {
  const props = {
    loading: false,
    topFilms: [
      { id: 1, title: 'Some title', poster_path: 'path to picture' },
      { id: 2, title: 'Another title', poster_path: 'path to another picture' },
    ],
  };

  it('Match its snapshot. Display Data from DataBse', () => {
    const TopFilmsComponent = shallow(<TopFilms {...props} />);
    expect(shallowToJson(TopFilmsComponent)).toMatchSnapshot();
  });

  it('Match its snapshot. Display ...Loading ', () => {
    const nextProps = {
      ...props,
      loading: true,
    };
    const TopFilmsComponent = shallow(<TopFilms {...nextProps} />);
    expect(shallowToJson(TopFilmsComponent)).toMatchSnapshot();
  });

  it('Tag content must be...Loading ', () => {
    const nextProps = {
      ...props,
      loading: true,
    };
    const TopFilmsComponent = shallow(<TopFilms {...nextProps} />);
    expect(TopFilmsComponent.find('li').text()).toEqual('...Loading');
  });

  it('Display Two films', () => {
    const TopFilmsComponent = shallow(<TopFilms {...props} />);
    expect(TopFilmsComponent.find('img')).toHaveLength(2);
  });
});
