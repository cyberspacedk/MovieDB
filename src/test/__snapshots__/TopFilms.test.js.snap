// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TopFilms Match its snapshot. Display ...Loading  1`] = `
<div>
  <styled.h2>
    Top Week chart 
  </styled.h2>
  <styled.ul>
    <li>
      ...Loading
    </li>
  </styled.ul>
</div>
`;

exports[`Component: TopFilms Match its snapshot. Display Data from DataBse 1`] = `
<div>
  <styled.h2>
    Top Week chart 
  </styled.h2>
  <styled.ul>
    <styled.li
      key="1"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w200path to picture"
      />
      <h2>
         
        Some title
         
      </h2>
    </styled.li>
    <styled.li
      key="2"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w200path to another picture"
      />
      <h2>
         
        Another title
         
      </h2>
    </styled.li>
  </styled.ul>
</div>
`;

exports[`Component: TopFilms should match its snapshot. Display ...Loading  1`] = `
<div>
  <styled.h2>
    Top Week chart 
  </styled.h2>
  <styled.ul>
    <li>
      ...Loading
    </li>
  </styled.ul>
</div>
`;

exports[`Component: TopFilms should match its snapshot. Display Data from DataBse 1`] = `
<div>
  <styled.h2>
    Top Week chart 
  </styled.h2>
  <styled.ul>
    <styled.li
      key="1"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w200path to picture"
      />
      <h2>
         
        Some title
         
      </h2>
    </styled.li>
    <styled.li
      key="2"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w200path to another picture"
      />
      <h2>
         
        Another title
         
      </h2>
    </styled.li>
  </styled.ul>
</div>
`;
